### PR TITLE
feat(import-bundle): Support endoScript

### PR DESCRIPTION
**Pull request by kriskowal**
## Description

Adds support for `endoScript` to the `importBundle` runtime.

### Security Considerations

None.

### Scaling Considerations

None.

### Documentation Considerations

This change makes safe the assumption that all bundles produced by `bundleSource` can be consumed by `importBundle`.

- NEWS
- README

### Testing Considerations

Unit tests included.

### Compatibility Considerations

None.

### Upgrade Considerations

None.

### CODE REVIEW

File: packages/import-bundle/src/index.js [hunk 1]
```diff
@@ -98,17 +101,27 @@ export async function importBundle(bundle, options = {}, powers = {}) {
...
-  const actualSource = `(${source})\n${sourceMap || ''}`;
-  const namespace = c.evaluate(actualSource)(filePrefix);
-  // namespace.default has the default export
-  return namespace;
+  compartment = new CompartmentToUse(endowments, {}, { transforms });
+  harden(compartment.globalThis);
+  const result = compartment.evaluate(source);
+  if (moduleFormat === 'endoScript') {
+    // The completion value of an 'endoScript' is the namespace.
+    // This format does curry the filePrefix.
```
Comment (warner) [inline packages/import-bundle/src/index.js:117]: s/does/does not/ ? or did you mean the format holds the filePrefix internally somehow?

File: packages/import-bundle/test/import-bundle.test.js [hunk 2]
```diff
@@ -152,6 +161,20 @@ test('inescapable global properties, nested evaluate format', async t => {
...
+    url.fileURLToPath(new URL('export-inescapable-global.js', import.meta.url)),
+    'endoScript',
+  );
+
+  const ns = await importBundle(bundle, {
+    inescapableGlobalProperties: {
+      inescapableGlobalValue: 42,
+    },
+  });
+  t.is(ns.default, 42);
```
Comment (warner) [inline packages/import-bundle/test/import-bundle.test.js:175]: Not really part of this PR, but should these tests exercise the "inescapability" of `inescapableGlobalProperties` ? Or did we decide that only the lexical-scope additions needed to be inescapable?

**Review by warner — APPROVED** [review]
LGTM, modulo the two questions

File: packages/import-bundle/test/import-bundle.test.js (hunk 2 above)
Comment (warner) [reply packages/import-bundle/test/import-bundle.test.js:175]: looks like transforms being inescapable might be important too

Comment (kriskowal) [reply packages/import-bundle/test/import-bundle.test.js:175]: The test fixture is a module that runs in a Compartment and creates a child Compartment without options. So, all of these are verifying inescapability at least one level down.
```js
export default new Compartment().evaluate('inescapableGlobalValue');
```
